File: 1456-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance/1456-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance.py
class Solution:
    def findTheCity(self, n: int, edges: List[List[int]], distanceThreshold: int) -> int:
        dist = [[int(1e9)]*n for _ in range(n)]
        for i in range(n):
            dist[i][i] = 0
        for i,j,w in edges:
            dist[i][j] = w
            dist[j][i] = w
        
        for k in range(n):
            for i in range(n):
                for j in range(n):
                    dist[i][j] = min(dist[i][j],dist[i][k]+dist[k][j])
                
        mini = int(1e9)
        m = -1
        for i in range(n):
            cnt = 0
            for j in range(n):
                if(i!=j and dist[i][j]<=distanceThreshold):
                    cnt+=1
            if(cnt<=mini):
                mini = cnt
                m = i
        return m

File: 1456-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance/1456-find-the-city-with-the-smallest-number-of-neighbors-at-a-threshold-distance.cpp
class Solution {
public:
    int findTheCity(int n, vector<vector<int>>& edges, int distanceThreshold) {
        vector<vector<int>> adj(n,vector<int>(n,INT_MAX));
        for(auto i : edges){
            adj[i[0]][i[1]] = i[2];
            adj[i[1]][i[0]] = i[2];
        }

        for(int k = 0;k<n;k++){
            for(int i = 0;i<n;i++){
                for(int j =0;j<n;j++){
                    if(adj[i][k]!=INT_MAX && adj[k][j]!=INT_MAX)
                    adj[i][j] = min(adj[i][j],adj[i][k]+adj[k][j]);
                }
            }
        }
        int maxi = INT_MAX;
        int res;
        for(int i = n-1;i>=0;i--){
            int count = 0;
            for(int j=0;j<n;j++){
                if(i!=j && adj[i][j] <=distanceThreshold){
                    count++;
                }
            }
            if(count<maxi){
                maxi = count;
                res = i;
            }
        }
        return res;
    }
};